policy-test/tests/outbound_api_app_protocol.rs

```diff
@@ -48,6 +47,91 @@ async fn opaque_parent() {
     test::<k8s::Service>().await;
 }
 
+#[cfg(feature = "gateway-api-experimental")]
+#[tokio::test(flavor = "current_thread")]
+async fn unknown_app_protocol_parent() {
```

Comment:
:pray: i love that you've added some additional test coverage as a part of this change. thank you @adleong!